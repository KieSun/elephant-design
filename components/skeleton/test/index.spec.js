import React from 'react'
import { mount, render } from 'enzyme'
import Skeleton from '../index'

describe('Skeleton', () => {
  it('the element when has not props', () => {
    const wrapper = mount(<Skeleton />)

    expect(wrapper.find('.eleph-skeleton').length).toBe(1)
    expect(wrapper.find('.eleph-skeleton__avatar').length).toBe(1)
    expect(wrapper.find('.eleph-skeleton__avatar--circle').length).toBe(1)
    expect(wrapper.find('.eleph-skeleton__title').length).toBe(1)
    expect(wrapper.find('.eleph-skeleton__paragraph > div').length).toBe(3)
  })

  it('affect the element when set props', () => {
    const wrapper = mount(
      <Skeleton
        animation={false}
        avatar={false}
        paragraph={false}
        title={false}
      >
        <div className="skeleton-children">loading completed</div>
      </Skeleton>
    )

    expect(wrapper.find('.eleph-skeleton').length).toBe(1)
    expect(wrapper.find('.eleph-skeleton__avatar').length).toBe(0)
    expect(wrapper.find('.eleph-skeleton__title').length).toBe(0)
    expect(wrapper.find('.eleph-skeleton__paragraph > div').length).toBe(0)
    expect(wrapper.find('.skeleton-children').length).toBe(0)

    const fourRowsWrapper = mount(
      <Skeleton paragraph={{ rows: 4 }} avatar={{ shape: 'square' }} />
    )
    expect(wrapper.find('.eleph-skeleton__avatar--circle').length).toBe(0)
    expect(
      fourRowsWrapper.find('.eleph-skeleton__paragraph > div').length
    ).toBe(4)

    const showChildrenWrapper = mount(
      <Skeleton loading={false}>
        <div className="skeleton-children">loading completed</div>
      </Skeleton>
    )
    expect(showChildrenWrapper.find('.eleph-skeleton').length).toBe(0)
    expect(showChildrenWrapper.find('.skeleton-children').length).toBe(1)
  })

  it("should render skeleton when it's loading", () => {
    const wrapper = render(
      <Skeleton>
        <div className="skeleton-children">loading completed</div>
      </Skeleton>
    )
    expect(wrapper).toMatchSnapshot('is-loading')
  })

  it("should render children when it's not loading", () => {
    const wrapper = render(
      <Skeleton loading={false}>
        <div className="skeleton-children">loading completed</div>
      </Skeleton>
    )
    expect(wrapper).toMatchSnapshot('not-loading')
  })
})
