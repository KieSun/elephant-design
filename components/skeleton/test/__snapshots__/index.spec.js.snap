// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skeleton should render children when it's not loading: not-loading 1`] = `
<div
  class="skeleton-children"
>
  loading completed
</div>
`;

exports[`Skeleton should render skeleton when it's loading: is-loading 1`] = `
<section
  class="eleph-skeleton eleph-skeleton--animation"
>
  <div
    class="eleph-skeleton__aside"
  >
    <div
      class="eleph-skeleton__avatar eleph-skeleton__avatar--circle"
    />
  </div>
  <div
    class="eleph-skeleton__content"
  >
    <h3
      class="eleph-skeleton__title"
    />
    <div
      class="eleph-skeleton__paragraph"
    >
      <div />
      <div />
      <div />
    </div>
  </div>
</section>
`;
